python/hail/docs/tutorials/03-expressions.ipynb

```diff
@@ -33,39 +98,65 @@
     }
    },
    "source": [
-    "# Eager Evaluation\n",
+    "### What is an Expression?\n",
     "\n",
-    "Python and R use **eager** evaulation.\n",
+    "Data types in Hail are represented by [expression](https://hail.is/docs/devel/expressions.html#expressions) classes. Each data type has its own expression class. For example, an integer of type `tint32` is represented by an `Int32Expression`. \n",
     "\n",
-    "When you enter an expression, the result is computed immediately and stored."
+    "We can construct an integer expression in Hail with the [int32](https://hail.is/docs/devel/functions/constructors.html?highlight=int32#hail.expr.functions.int32) function."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "slideshow": {
-     "slide_type": "fragment"
-    }
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
-    "1 + 2"
+    "hl.int32(3)"
    ]
   },
   {
    "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `literal` function will convert a Python object into a Hail expression. Let's try it out on a Python list."
```

Comment:
I think we should add links to the docs for the literal, value, and show functions.